### ISSUE TITLE
(optional) null should be part of matchedData

### ISSUE THREAD

**Issue by danger89** [express-validator/express-validator#568]
Hi,

**TLDR:** `matchedData()` is not returning my body field with JSON value of  `null`, while I try to validate the data with the check API.

I don't know if I use your validator incorrect or if it's a feature request.. but..
I use your validator for my backend app before sending the JSON object further to the database (eg. MariaDB).

I have a database field what either can be of type integer or have the value of `NULL`. So I tried to use your validator in this way (at least this is what I thought should work):
```js
body('journal_bank_id').optional({nullable: true}).isInt()
```

However, I'm also using the filter `matchedData()` function in order to guarantee that my data I put in my database is fully validated first by your validator package :+1: 
```js
object = matchedData(req)
```

Now, when I try to request an API call to the Express backend using your validator with the HTTP body:
```json
{"journal_bank_id": null}
```
**Expected situation:** I would expect the JSON object is now validated correctly and the journal_bank_id with null value becomes part of the returned object from `matchedData()`. Meaning the MySQL query will set the corresponding field to NULL accordantly. So basically I would expect `matchedData()` returns:
`{"journal_bank_id": null}`

**Actual situation:** `matchedData()` is **_not_** returning `journal_bank_id` with value `null` at all. Meaning my database field can't be set to NULL because of your validator. 

Do I miss something here? Should I maybe not use optional()? I couldn't find `isNull()`.. I have no clue anymore..

Thanks in advance!

Kind regards,
Melroy van den Berg

**Comment by danger89** [comment]
Like previous version supported: `.allow(null)` if I'm not mistaken..

**Comment by gustavohenke** [comment]
Thanks for reporting this, @danger89. Will need to add an optional to include optional data, otherwise could break people's applications.

**Comment by danger89** [comment]
Sure whatever is needed

**Comment by danger89** [comment]
@gustavohenke 
The interface says `filterOptionals` not `includeOptionals`. SO what is it?

I'm getting:
```
error TS2345: Argument of type '{ includeOptionals: boolean; }' is not assignable to parameter of type 'Partial<MatchedDataOptions>'.
  Object literal may only specify known properties, and 'includeOptionals' does not exist in type 'Partial<MatchedDataOptions>'.```

**Comment by danger89** [comment]
And waiting on a new release :+1:

**Comment by gustavohenke** [comment]
The interface is wrong

**Comment by JoseExposito** [comment]
First, thank you very much for the PR @gustavohenke

Any estimated release date? I'm also affetected by this issue. See #605

**Comment by gustavohenke** [comment]
I have been unable to dedicate much time to this project lately, that's why I haven't released it yet.
I don't expect to add anything else to the [already quite big log of changes of an eventual `v5.3.0`](https://github.com/express-validator/express-validator/compare/v5.2.0...master).

**Comment by JoseExposito** [comment]
Thank you very much for your answer and effort put in the project.

Looking forward for the next release and hopping you manage to spare some time. This kind of projects can be very time consuming, so let me know if you need help with the release.

**Comment by GitStorageOne** [comment]
Current logic is not correct.
ValidationSchema contain:
```
optional: {
  options: { nullable: true }
}
```
matchedData() removes fields with `null` value. 
null is proper data type, that present in JSON and JS, and should not be dropped when `nullable: true`.

If we pass `includeOptionals`, we would be able to receive null values, but this adds undesired effect, because if datashape is not include optional field at all, matchedData will add it with undefined value.

Expected behavior is to pass `null` without `includeOptionals` option when `nullable: true`.
So, when client not specify optional field, we will not see it, as expected.
When client send null, we will get null as expected and database will accept that type.
That would be correct logic, but may break some user projects.

Current workaround is to set includeOptionals = true and manually delete all undefined fields which is not great.
